### DISCUSSION TITLE
Floor heating wind up parameters

### DISCUSSION THREAD

**Discussion by ollila** [veista/nilan#126] [General]
I have been using number.nilan_central_heating_min_supply_temperature as control parameter for low cost spot-price heating with good success. Now I have been playing around with the same entity to utilize excess electricity from solar panels within 15 mins net balance window. 

The control through this entity works but is sub-optimal especially when balance gets negative close to end of window. As a practical example if balance gets negative 2-3 min before window raising the number.nilan_central_heating_min_supply_temperature to 32 C will not provide load fast enough to utilize the excess balance. Does anyone know how the central heating control strategy works? There is "Integration time" parameter for supply air temperature suggesting unit is capable to PID- or PI-based control. Preferably I would like to either increase P- value or decrease integration time for central heating. Another strategy would be increase set point from 32C assuming there is P-value taken into account in control.

**Comment by ollila** [comment]
The unit in questions is post-facelift EC9 with Firmware: 1.6.52 and Hardware: 16.

**Comment by veista** [comment]
Is there a parameter in the HMI of your device that you want implemented in the integration?

Is there a question here, or are you just sharing your setup?

**Reply by ollila** [reply]
Let me rephrase: The integration prodides entity number.nilan_supply_air_pid_integration_time for supply air but not for central heating. There is similar setting for central heating in HMI (see pic) which is not implemented by the integration. Question: Could this be implemented by the integration? I would be interested also to hear experiences using the integration time for faster control reaction if someone has experienced with it.
![WhatsApp Image 2024-04-08 at 13 20 41](https://github.com/veista/nilan/assets/113829131/31972065-8296-43e5-a0f6-a14cb4043706)

**Reply by veista** [reply]
This is possible to add depending on your modbus version. Please open an issue and follow the instructions in readme.

**Reply by ollila** [reply]
Playing with the setting and reading Nilan documentation the existing integration time -entity is this central heating integration time and there is no "Supply air" connected integration time setting. I can still raise this as issue as mislabeling if that's the best way to resolve.
